validations/src/validations.rs

```diff
@@ -509,7 +511,16 @@ pub fn validate_data_request_output(
     }
 
     if active_wips.wip0022() {
-        let reward_collateral_ratio = request.collateral / request.witness_reward;
+        let collateral = if request.collateral == 0 {
+            collateral_minimum
+        } else {
+            request.collateral
+        };
```

Comment:
This is a bit strange, because there's a gap from 0 to collateral_minimum, so I assume you actually wanted to cap it like this:

```suggestion
        let collateral = std::cmp::max(request.collateral, collateral_minimum);
```

Reply:
This is actually on purpose. Your suggestion changes the validation behavior.

If a data request with a 100 mWIT collateral requirement and 1 mWIT reward is sent into the network it is valid from the perspective of the `TransactionError::RewardTooLow` error (assuming a 100 to 1 ratio). Changing the collateral to `collateral_minimum` would result in the error being (incorrectly) thrown.

What I actually want to see is that the `TransactionError::InvalidCollateral` error is being thrown in the code following the data request output validation given that this is more informative to the end-user building the data request. AFAIK errors will act as a short-circuit and thus adopting your suggestion would prevent the 'correct' error from being thrown.

Reply:
> If a data request with a 100 mWIT collateral requirement and 1 mWIT reward is sent into the network it is valid from the perspective of the `TransactionError::RewardTooLow` error (assuming a 100 to 1 ratio). Changing the collateral to `collateral_minimum` would result in the error being (incorrectly) thrown.

Indeed I'm totally expecting a `TransactionError::InvalidCollateral` in this case because 1 mWit < `collateral_minimum`.